src/ucai/core/databricks.py

```diff
@@ -111,6 +115,55 @@ def extract_function_name(sql_body: str) -> str:
     )
 
 
+def retry_on_session_expiration():
+    """
+    Decorator to retry a method upon session expiration errors with exponential backoff.
+
+    Args:
+        max_attempts (int): Maximum number of retry attempts.
+        base_delay (int): Base delay in seconds for exponential backoff.
+        max_delay (int): Maximum delay in seconds between retries.
+    """
+    max_attempts = int(UCAI_DATABRICKS_SESSION_RETRY_MAX_ATTEMPTS.get())
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            for attempt in range(1, max_attempts + 1):
+                try:
+                    return func(self, *args, **kwargs)
+                except Exception as e:
+                    error_message = str(e)
+                    if (
+                        "session_id is no longer usable" in error_message
+                        or "BAD_REQUEST: session_id is no longer usable" in error_message
```

Comment:
```suggestion
                        "session_id is no longer usable" in error_message
```
Do we need `or`?

Reply:
Not sure what the heck I was thinking here? lol... long day.